### PR TITLE
Upgrade to React Hook Form version 7

### LINKED ISSUE #812: Update react-hook-form v7
More information can be found in the [migration guide](https://react-hook-form.com/migrate-v6-to-v7/). This is quite a large update as the API surface has changed significantly.

### PR DESCRIPTION
**Pull request by jonkoops**
This is a work-in-progress draft to complete the migration to React Hook Form version 7. Closes #812.

### CODE REVIEW

File: apps/account-ui/src/resources/EditTheResource.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const EditTheResource = ({
   const { t } = useTranslation();
   const { addAlert, addError } = useAlerts();
 
-  const form = useForm<FormValues>({ shouldUnregister: false });
```
Comment (edewit) [inline apps/account-ui/src/resources/EditTheResource.tsx:30]: why this change?

File: apps/admin-ui/src/components/form-access/FormAccess.test.tsx [hunk 2]
```diff
@@ -1,64 +0,0 @@
-/**
```
Comment (edewit) [inline apps/admin-ui/src/components/form-access/FormAccess.test.tsx:1]: does this test no longer work?

**Review by edewit — APPROVED** [review]
:tada:

File: apps/account-ui/src/resources/EditTheResource.tsx (hunk 1 above)
Comment (jonkoops) [reply apps/account-ui/src/resources/EditTheResource.tsx:30]: The default value of `shouldUnregister` is now `false`, so there is no need to add this any longer.

File: apps/admin-ui/src/components/form-access/FormAccess.test.tsx (hunk 2 above)
Comment (jonkoops) [reply apps/admin-ui/src/components/form-access/FormAccess.test.tsx:1]: Yeah, had some issues getting this to work properly.
